File: backend/internal/model/project.go
package model

import "time"

type Project struct {
    ID              uint      `db:"id" json:"id"`
    Slug            string    `db:"slug" json:"slug"`
    FullName        string    `db:"full_name" json:"full_name"`
    Description     *string   `db:"description" json:"description,omitempty"`
    StartedAt       *string   `db:"started_at" json:"started_at,omitempty"`        // ← fixed
    EndedAt         *string   `db:"ended_at" json:"ended_at,omitempty"`            // ← fixed
    StartedAtJalali *string   `db:"started_at_jalali" json:"started_at_jalali,omitempty"`
    EndedAtJalali   *string   `db:"ended_at_jalali" json:"ended_at_jalali,omitempty"`
    Employer        string    `db:"employer" json:"employer"`
    IsVisible       bool      `db:"is_visible" json:"is_visible"`
    DisplayOrder    int       `db:"display_order" json:"display_order"`
    CreatedBy       string    `db:"created_by" json:"created_by"`
    CreatedAt       time.Time `db:"created_at" json:"created_at"`
    UpdatedAt       time.Time `db:"updated_at" json:"updated_at"`
}

File: frontend/types/Project.ts
export interface Project {
  id: number;
  slug: string;
  full_name: string;
  description: string | null;
  started_at: string | null;
  ended_at: string | null;
  started_at_jalali: string | null;
  ended_at_jalali: string | null;
  employer: string;
  is_visible: boolean;
  display_order: number;
  created_by: string;
  created_at: string;
  updated_at: string;
  images: Image[];
}

export interface Image {
  id: number;
  project_id: number;
  image_url: string;
  alt_text: string;
  sort_order: number;
  created_at: string;
}

export interface PaginatedProjects {
  data: Project[];
  total: number;
  page: number;
  per_page: number;
  total_pages: number;
}

// For updates
export interface UpdateProjectRequest {
  full_name?: string;
  description?: string;
  started_at?: string;
  ended_at?: string;
  started_at_jalali?: string;
  ended_at_jalali?: string;
  employer?: string;
  is_visible?: boolean;
  display_order?: number;
}

export interface UpdateImageRequest {
  alt_text?: string;
  sort_order?: number;
}